### PR TITLE
pmixp: added the direct connect authentification

### PR DESCRIPTION
**Pull request by karasevb**

### CODE REVIEW

File: src/plugins/mpi/pmix/pmixp_server.c [hunk 1]
```diff
@@ -488,6 +490,62 @@ void pmixp_server_cleanup(void)
...
+static int _auth_cred_create(Buf buf) {
+	void *auth_cred = NULL;
+	char *auth_info = slurm_get_auth_info();
+	int rc = SLURM_SUCCESS;
+
+	auth_cred = g_slurm_auth_create(auth_info);
+	if (NULL == auth_cred) {
+		rc = g_slurm_auth_errno(NULL);
+		PMIXP_ERROR("Creating authentication credential: %s",
+		g_slurm_auth_errstr(rc));
```
Comment (karasevb) [inline src/plugins/mpi/pmix/pmixp_server.c]: offset

File: src/plugins/mpi/pmix/pmixp_server.c [hunk 2]
```diff
@@ -488,6 +490,62 @@ void pmixp_server_cleanup(void)
...
+		goto exit;
+	}
+	rc = g_slurm_auth_pack(auth_cred, buf);
+	if (SLURM_SUCCESS != rc) {
+		PMIXP_ERROR("Packing authentication credential: %s",
+		g_slurm_auth_errstr(g_slurm_auth_errno(auth_cred)));
```
Comment (karasevb) [inline src/plugins/mpi/pmix/pmixp_server.c]: offset

File: src/plugins/mpi/pmix/pmixp_server.c [hunk 3]
```diff
@@ -488,6 +490,62 @@ void pmixp_server_cleanup(void)
...
+
+static int _auth_cred_verify(Buf buf) {
+	void *auth_cred = NULL;
+	char *auth_info = NULL;
+	int rc = SLURM_SUCCESS;
+
+	auth_cred = g_slurm_auth_unpack(buf);
+	if (NULL == auth_cred) {
+		PMIXP_ERROR("Unpacking authentication credential: %s",
+		g_slurm_auth_errstr(g_slurm_auth_errno(NULL)));
```
Comment (karasevb) [inline src/plugins/mpi/pmix/pmixp_server.c]: offset

File: src/plugins/mpi/pmix/pmixp_server.c [hunk 4]
```diff
@@ -488,6 +490,62 @@ void pmixp_server_cleanup(void)
...
+		return SLURM_FAILURE;
+	}
+	auth_info = slurm_get_auth_info();
+	rc = g_slurm_auth_verify(auth_cred, auth_info);
+	if (rc) {
+		PMIXP_ERROR("Verifying authentication credential: %s",
+		g_slurm_auth_errstr(g_slurm_auth_errno(auth_cred)));
```
Comment (karasevb) [inline src/plugins/mpi/pmix/pmixp_server.c]: offset

File: src/plugins/mpi/pmix/pmixp_server.c [hunk 5]
```diff
@@ -488,6 +490,62 @@ void pmixp_server_cleanup(void)
...
+ * --------------------- Authentication functionality -------------------
+ */
+
+static int _auth_cred_create(Buf buf) {
+	void *auth_cred = NULL;
+	char *auth_info = slurm_get_auth_info();
+	int rc = SLURM_SUCCESS;
+
+	auth_cred = g_slurm_auth_create(auth_info);
+	if (NULL == auth_cred) {
```
Comment (artpol84) [inline src/plugins/mpi/pmix/pmixp_server.c]: `if (!auth_cred) {`

File: src/plugins/mpi/pmix/pmixp_server.c [hunk 6]
```diff
@@ -488,6 +490,62 @@ void pmixp_server_cleanup(void)
...
+		rc = g_slurm_auth_errno(NULL);
+		PMIXP_ERROR("Creating authentication credential: %s",
+			    g_slurm_auth_errstr(rc));
+		goto exit;
+	}
+	rc = g_slurm_auth_pack(auth_cred, buf);
+	if (SLURM_SUCCESS != rc) {
```
Comment (artpol84) [inline src/plugins/mpi/pmix/pmixp_server.c]: Same here,
assume that `SLURM_SUCCESS` is always == 0. So you can just check
`if (rc) {

File: src/plugins/mpi/pmix/pmixp_server.c [hunk 7]
```diff
@@ -488,6 +490,62 @@ void pmixp_server_cleanup(void)
...
+		PMIXP_ERROR("Packing authentication credential: %s",
+			    g_slurm_auth_errstr(g_slurm_auth_errno(auth_cred)));
+		goto exit;
+	}
+
+exit:
+	xfree(auth_info);
+	if (auth_cred) {
```
Comment (artpol84) [inline src/plugins/mpi/pmix/pmixp_server.c:518]: :+1: :)

File: src/plugins/mpi/pmix/pmixp_server.c [hunk 8]
```diff
@@ -488,6 +490,62 @@ void pmixp_server_cleanup(void)
...
+	return rc;
+}
+
+static int _auth_cred_verify(Buf buf) {
+	void *auth_cred = NULL;
+	char *auth_info = NULL;
+	int rc = SLURM_SUCCESS;
+
+	auth_cred = g_slurm_auth_unpack(buf);
+	if (NULL == auth_cred) {
```
Comment (artpol84) [inline src/plugins/mpi/pmix/pmixp_server.c]: simplify

File: src/plugins/mpi/pmix/pmixp_server.c [hunk 9]
```diff
@@ -488,6 +490,62 @@ void pmixp_server_cleanup(void)
...
+		PMIXP_ERROR("Unpacking authentication credential: %s",
+			    g_slurm_auth_errstr(g_slurm_auth_errno(NULL)));
+		return SLURM_FAILURE;
+	}
+	auth_info = slurm_get_auth_info();
+	rc = g_slurm_auth_verify(auth_cred, auth_info);
+	if (rc) {
```
Comment (artpol84) [inline src/plugins/mpi/pmix/pmixp_server.c:537]: :+1:

File: src/plugins/mpi/pmix/pmixp_server.c [hunk 10]
```diff
@@ -488,6 +490,62 @@ void pmixp_server_cleanup(void)
...
+		PMIXP_ERROR("Verifying authentication credential: %s",
+			    g_slurm_auth_errstr(g_slurm_auth_errno(auth_cred)));
+		g_slurm_auth_destroy(auth_cred);
+		return rc;
+	}
+	xfree(auth_info);
+	g_slurm_auth_destroy(auth_cred);
+
```
Comment (artpol84) [inline src/plugins/mpi/pmix/pmixp_server.c]: - extra line

**Pull request closed without merging**